includes/payments/functions.php

```diff
@@ -223,15 +223,9 @@ function edd_delete_purchase( $payment_id = 0, $update_customer = true, $delete_
 	global $edd_logs;
 
 	$payment   = new EDD_Payment( $payment_id );
-	$downloads = $payment->downloads;
-
-	if ( is_array( $downloads ) ) {
-		// Update sale counts and earnings for all purchased products
-		foreach ( $downloads as $download ) {
-			edd_undo_purchase( $download['id'], $payment_id );
-		}
-	}
 
+	// Update sale counts and earnings for all purchased products
+	edd_undo_purchase( false, $payment_id );
```

Comment:
First parameter is expected to be the download ID. Since we determined that this function was a bit broken, we should mark this as deprecated and force it to false inside the function.